### DISCUSSION TITLE
Enable GitHub apps to read images in private repositories?

### DISCUSSION THREAD

**Discussion by tomasreimers** [community/community#57488] [Apps, API and Webhooks]
### Select Topic Area

Product Feedback

### Body

GitHub recently changed image visibility in private respositories: https://github.blog/changelog/2023-05-09-more-secure-private-attachments/

Before this change, when a user uploaded an image to GH, the URL of the image was: "https://user-uploads.github.com/..." After this change, uploading images to PRs results in a URL like: https://github.com/xxx/yyy/assets/123/456/ 

Per @vtbassmatt, I have found I can get this path to work by proxying the request and including an `Authorization: token XXX` header (https://github.com/orgs/community/discussions/54551#discussioncomment-5851902). However, this only works with OAUTH and PAT tokens. It does not work with app user tokens.

This prevents Github apps from being able to load private asset paths (including Github actions). 

Could we update the API to support user app tokens as well? (users oauthed into GH, but through an app)

**Comment by tommoor** [comment]
This limitation is completely hobbling our integration, would love to see it addressed.
